### PR TITLE
[SYCL] Renaming fixed_size_group to chunk

### PR DESCRIPTION
**Pull request by AndreiZibrov**
First part of changes in order of https://github.com/intel/llvm/pull/14604:
fixed_size_group -> chunk

### CODE REVIEW

File: sycl/include/sycl/ext/oneapi/experimental/chunk.hpp [hunk 1]
```diff
@@ -26,18 +26,18 @@ namespace sycl {
...
 #ifdef __SYCL_DEVICE_ONLY__
-[[__sycl_detail__::__uses_aspects__(sycl::aspect::ext_oneapi_fixed_size_group)]]
+[[__sycl_detail__::__uses_aspects__(sycl::aspect::ext_oneapi_chunk)]]
 #endif
 inline std::enable_if_t<sycl::is_group_v<std::decay_t<Group>> &&
                             std::is_same_v<Group, sycl::sub_group>,
-                        fixed_size_group<PartitionSize, Group>>
-get_fixed_size_group(Group group);
+                        chunk<PartitionSize, Group>>
+get_chunk(Group group);
```
Comment (AlexeySachkov) [inline sycl/include/sycl/ext/oneapi/experimental/chunk.hpp:38]: [`sycl_ext_oneapi_non_uniform_groups`](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_non_uniform_groups.asciidoc) does not define `get_chunk`, I suppose you meant `chunked_partition`

File: sycl/include/sycl/ext/oneapi/experimental/chunk.hpp [hunk 2]
```diff
@@ -26,18 +26,18 @@ namespace sycl {
...
 
-template <size_t PartitionSize, typename ParentGroup> class fixed_size_group {
+template <size_t PartitionSize, typename ParentGroup> class chunk {
```
Comment (AlexeySachkov) [inline sycl/include/sycl/ext/oneapi/experimental/chunk.hpp:40]: We should also leave a `TODO` comment to implement `operator fragment<ParentGroup>() const;` once `fragment` is ready

File: sycl/test-e2e/NonUniformGroups/fixed_size_group.cpp
Comment (AlexeySachkov) [inline sycl/test-e2e/NonUniformGroups/fixed_size_group.cpp:1]: We should probably rename the file as well. Also applies to `fixed_size_group_algorithms.cpp`

File: sycl/include/sycl/ext/oneapi/experimental/chunk.hpp (hunk 2 above)
Comment (AlexeySachkov) [inline sycl/include/sycl/ext/oneapi/experimental/chunk.hpp:40]:
```suggestion
template <size_t ChunkSize, typename ParentGroup> class chunk {
```

That's rather minor, but I think that it is better to be aligned in the terminology we use

**Review by AlexeySachkov — CHANGES_REQUESTED** [review]

**Comment by steffenlarsen** [comment]
I am concerned about doing the changes in https://github.com/intel/llvm/pull/14604 gradually. I do not mind having multiple patches for them, but they should go in around the same time. We also need to make sure we coordinate these changes with changes to https://github.com/KhronosGroup/SYCL-CTS/tree/SYCL-2020/tests/extension/oneapi_non_uniform_groups.

**Pull request closed without merging**